Fix bug in process_group_name when there is duplicate pgs

xw285cornell opened the pull request:
Summary: with the new c10d API, we don't need all ranks to call new_group. Integrate with the new API, so that every rank just call new_group 3 times, with a local barrier with the members within the group.


Reviewed By: xunnanxu, eeggl

Differential Revision: D45315615

eeggl commented on test/distributed/test_c10d_common.py:
```diff
@@ -1417,6 +1417,43 @@ def _test_new_group_local_sync_sanity_check(self, backend):
 
+    def _test_new_group_local_sync_duplidate_pg(self, backend):
+        """
+        We should support users create multiople PGs with the same set of
```
`s/multiople/multiple/`

kumpera reviewed (approved):
Thanks for the fix!

kumpera commented:
@pytorchmergebot merge

xw285cornell commented:
sorry - I should have a separate PR for it (this is with the integration changes in fair scale). I'll land it from inside

xw285cornell committed:
Fix bug in process_group_name when there is duplicate pgs (#100518)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/100518

with the new c10d API, we don't need all ranks to call new_group. Integrate with the new API, so that every rank just call new_group 3 times, with a local barrier with the members within the group.

Test Plan: https://www.internalfb.com/mlhub/pipelines/runs/mast/torchx_hpc-xlformers_chinch70B_4096_xdwang_0426190441?env=PRODUCTION&job_name=torchx_hpc-xlformers_chinch70B_4096_xdwang_0426190441

Reviewed By: xunnanxu, eeggl

Differential Revision: D45315615

fbshipit-source-id: b499a2e085722cc429a3ad34b9da1093decded46

xw285cornell commented:
@pytorchbot  merge
